git/test/test_submodule.py

```diff
@@ -660,6 +661,24 @@ def test_add_empty_repo(self, rwdir):
                                   url=empty_repo_dir, no_checkout=checkout_mode and True or False)
         # end for each checkout mode
 
+    @with_rw_directory
+    def test_list_only_valid_submodules(self, rwdir):
```

Comment:
Previously this failed due to the exception above, but maybe there 'list submodule' code itself can protect against said exception. That way this test will pass, without breaking the currently tested behaviour.